main-english comments.py

```diff
@@ -1,180 +1,181 @@
-    data_array = np.asarray(originalImage)
-    data_Format = len(data_array[0][0])
-    width, height = originalImage.size
-
-    # We change the array so that it has new dimensions
-    # -> before we had a 3d array, now we have a 2d array
-    # we use .reshape to achieve this
-    data_array_sortContinuously = data_array.reshape(width * height, data_Format).copy()
-    quickSortHelp(data_array_sortContinuously, howBright)
-    # after having sorted the array, we turn it back into a 3d array
-    converted_data_array = data_array_sortContinuously.reshape(height, width, data_Format)
-    converted_image_continously = Image.fromarray(converted_data_array, "RGBA"[:data_Format])
-    return converted_image_continously
-
-#### Sorting by pixels
-original = Image.open('Bilder/Beispiel.PNG')
-
-# Make the image smaller while keeping the proportions
-factor = 0.25
-width, height = original.size
-width, height = int(width * factor), int(height * factor)
-original = original.resize((width, height))
-
-def convert(originalImage):
-    original = originalImage
-    convertedImageRow = convertRow(originalImage)
-    print("Finished sorting each row")
-
-    convertedImageColumn = convertColumn(originalImage)
-    print("Finished sorting each colunmn")
-
-    convertedImageContinuously = convertContinuously(originalImage)
-    print("Finished sorting continuously")
-
-    # Displaying it in plotview
-    fig, axs = plt.subplots(2, 2)
-
-    axs[0][0].imshow(original)
-    axs[0][0].set_title("Orginal Image")
-    axs[0][0].axis('off')
-
-    axs[0][1].imshow(convertedImageRow)
-    axs[0][1].set_title("Sorted each Row")
-    axs[0][1].axis('off')
-
-    axs[1][0].imshow(convertedImageColumn)
-    axs[1][0].set_title("Sorted each Column")
-    axs[1][0].axis('off')
-
-    axs[1][1].imshow(convertedImageContinuously)
-    axs[1][1].set_title("Sorted Continuously")
-    axs[1][1].axis('off')
-
-    fig.tight_layout()
-    plt.show()
-
-convert(original)
+from PIL import Image 
+import numpy as np 
+import matplotlib.pyplot as plt 
+
+ 
+### The Quicksort Algorithm 
+# This is a helpermethod for the quickSort Algorithm 
+# We need to give the quickSort the parameters arr, low, high and comparator every time 
+# arr = is the array we want to sort 
+# low = the index of the first element in our array (NOT the value) 
+# high = the index of the last element in our array (NOT the value) 
+# comparator = a helpermethod which we use to decide how to compare stuff 
+ 
+# Every time we want to sort a picture, we have to pass on the above mentioned 4 parameters. To avoid code duplication, 
+# we created this helpermethod 
+ 
+# This way we only need to manually pass on the array and the comparator method, the helpermethod does the rest for us 
+ 
+def quickSortHelp(arr, comparator): 
+    quickSort(arr, 0, len(arr) - 1, comparator) 
+ 
+ 
+def quickSort(arr, low, high, comparator): 
+    if len(arr) == 1: 
+        return arr 
+    if low < high: 
```

Comment:
I think you could use an else here. If low > high then maybe you want to raise an exception or swap them.